src/main/scala/com/wavesplatform/Coordinator.scala

```diff
@@ -87,6 +88,7 @@ object Coordinator extends ScorexLogging with Instrumented {
                          settings: WavesSettings, featureProvider: FeatureProvider)
                         (newBlock: Block, local: Boolean): Either[ValidationError, BigInt] = measureSuccessful(blockProcessingTimeStats, {
     val newScore = for {
+      _ <- Either.cond(history.heightOf(newBlock.reference).exists(_ >= history.height() - 1), (), GenericError("Can process either new top block or current top block's competitor"))
```

Comment:
What if we received the same block? I think we should ignore it and this situation is not rare :)